packages/ember-glimmer/tests/integration/helpers/unbound-test.js

```diff
@@ -0,0 +1,545 @@
+        this._super(...arguments);
+      },
+
+      compute([value]) {
+        if (this.get('value')) {
+          this.removeObserver('value.firstName');
+        }
+        this.set('value', value);
+        this.addObserver('value.firstName', this, this.recompute);
+        return (value ? get(value, 'firstName').toUpperCase() : '');
+      }
+    });
+
+    this.registerHelper('-concatNames', {
+      destroy() {
+        this.teardown();
+        this._super(...arguments);
+      },
+      teardown() {
+        this.removeObserver('value.firstName');
+        this.removeObserver('value.lastName');
+      },
+      compute([value]) {
+        if (this.get('value')) {
+          this.teardown();
+        }
+        this.set('value', value);
+        this.addObserver('value.firstName', this, this.recompute);
+        this.addObserver('value.lastName', this, this.recompute);
+        return (value ? get(value, 'firstName') : '') + (value ? get(value, 'lastName') : '');
+      }
+    });
+
+    this.render(`{{-capitalizeName person}} {{unbound (-capitalizeName person)}} {{-concatNames person}} {{unbound (-concatNames person)}}`, {
+      person: EmberObject.create({
+        firstName: 'shooby',
+        lastName:  'taylor'
+      })
+    });
+
+    this.assertText('SHOOBY SHOOBY shoobytaylor shoobytaylor');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('SHOOBY SHOOBY shoobytaylor shoobytaylor');
+
+    this.runTask(() => set(this.context, 'person.firstName', 'sally'));
+
+    this.assertText('SALLY SHOOBY sallytaylor shoobytaylor');
+
+    this.runTask(() => set(this.context, 'person', EmberObject.create({
+      firstName: 'shooby',
+      lastName:  'taylor'
+    })));
+
+    this.assertText('SHOOBY SHOOBY shoobytaylor shoobytaylor');
+  }
+
+  ['@htmlbars should be able to render an unbound helper invocation in #each helper']() {
+    this.registerHelper('-capitalize', (params) => params[0].toUpperCase());
+
+    this.render(`{{#each people as |person|}}{{-capitalize person.firstName}} {{unbound (-capitalize person.firstName)}}{{/each}}`, {
+      people: emberA([
+        {
+          firstName: 'shooby',
+          lastName:  'taylor'
+        },
+        {
+          firstName: 'cindy',
+          lastName:  'taylor'
+        }
+      ])
+    });
+
+    this.assertText('SHOOBY SHOOBYCINDY CINDY');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('SHOOBY SHOOBYCINDY CINDY');
+
+    this.runTask(() => this.context.people.setEach('firstName', 'chad'));
+
+    this.assertText('CHAD SHOOBYCHAD CINDY');
+
+    this.runTask(() => set(this.context, 'people', emberA([
+      {
+        firstName: 'shooby',
+        lastName:  'taylor'
+      },
+      {
+        firstName: 'cindy',
+        lastName:  'taylor'
+      }
+    ])));
+
+    this.assertText('SHOOBY SHOOBYCINDY CINDY');
+  }
+
+  ['@htmlbars should be able to render an unbound helper invocation with bound hash options']() {
```

Comment:
Port of https://github.com/emberjs/ember.js/blob/bf067f7370c3c3381352d68ef5dfe55d1178c6f7/packages/ember-htmlbars/tests/helpers/unbound_test.js#L433-L458